File: src_python/06_interval_proporsi.py
import math
# 1. BAGIAN INPUT
# memasukkan nilai proporsi, alpha, dan n
input_proporsi = float(input("Masukkan proporsi: "))
input_alpha = float(input("Masukkan alpha: "))
input_n = int(input("Masukkan n: "))

# 2. DEFINISI FUNGSI UTAMA
def hitung_interval(proporsi, alpha, n):
    # Validasi jika angka proporsi tidak sesuai syarat
    if proporsi < 0 or proporsi > 1:
        return "Error: Proporsi sampel > 1 / < 0"
    
    # nilai z-score berdasarkan input persentase alpha
    if alpha == 10:
        z = 1.645
    elif alpha == 5:
        z = 1.96
    else:
        return "Error: alpha harus 5 atau 10"

    # Rumus Margin of Error dan Batas Interval    
    margin = z * math.sqrt((proporsi * (1 - proporsi)) / n)
    batas_bawah = proporsi - margin
    batas_atas = proporsi + margin
    # Mengembalikan hasil hitungan ke luar fungsi
    return batas_bawah, batas_atas 

# 3. MEMANGGIL FUNGSI DAN OUTPUT
hasilnya = hitung_interval(input_proporsi, input_alpha, input_n)
print("Hasil Interval:", hasilnya)

File: src_R/06_interval_proporsi.R
proporsi<-0.7
alpha<-10
n<-15

Interval <- function(proporsi, alpha, n) {
  if (proporsi < 0 || proporsi > 1) {
    return("Error: Proporsi sampel > 1 / < 0")
  }
  
  if (alpha == 10) {
    z <- 1.645
  } else if (alpha == 5) {
    z <- 1.96
  } else {
    return("Error: alpha harus 5 atau 10")
  }
  
  margin <- z * sqrt((proporsi * (1 - proporsi)) / n)
  batas_bawah <- proporsi - margin
  batas_atas <- proporsi + margin
  
  return(list(bawah = batas_bawah, atas = batas_atas)) 
}

hasilnya <- Interval(proporsi,alpha,n)
if (is.character(hasilnya)) {
  cat(hasilnya, "\n")
  } else {
  cat("Batas Bawah :", hasilnya$bawah, "\n")
  cat("Batas Atas  :", hasilnya$atas, "\n")
  }
}
  
